### PR TITLE
wl: do not set to null the surface and target at touch_on_up()

### PR DESCRIPTION
**Pull request by clopez**
On a multitouch screen, when several fingers are on the screen at the same time, if we set to null surface and target on touch_on_up() then we stop all the ongoing touch gestures and not only the one of the finger that caused the up event. That is becasue touch_on_motion() will refuse to process further events if surface or target are null.

Instead of doing that do not set to null this pointers on the touch_on_up() event. The value of those will be set again on the next touch_on_down() event in any case.

Related: https://github.com/Igalia/cog/issues/698
Related: https://bugs.webkit.org/show_bug.cgi?id=270516

### CODE REVIEW

**Review by psaavedra — APPROVED** [review]
This looks good to me

**Review by aperezdc — CHANGES_REQUESTED** [review]
I think it would be better to make `seat->touch_target` a weak reference, that way if the `CogViewport` gets destroyed before the touch-move and/or touch-up events arrive it will be set to `NULL` automatically and we avoid use-after-free of a stale pointer.

When assigning inside `touch_on_down` we want to use:

```diff
-    seat->touch_target = viewport;
+    g_set_weak_pointer(&seat->touch_target, viewport);
```

File: platform/wayland/cog-platform-wl.c [hunk 1]
```diff
@@ -844,7 +839,8 @@ touch_on_up(void *data, struct wl_touch *touch, uint32_t serial, uint32_t time,
 
     struct wpe_input_touch_event event = {seat->touch.points, 10, raw_event.type, raw_event.id, raw_event.time};
 
-    CogView *view = cog_viewport_get_visible_view((CogViewport *) viewport);
+    CogWlViewport *viewport = COG_WL_VIEWPORT(seat->touch_target);
```
[the remark this answers is not in the document]
Comment (clopez) [reply platform/wayland/cog-platform-wl.c:842]: It is already checked at the start of `touch_on_up()`

```
static void
touch_on_up(void *data, struct wl_touch *touch, uint32_t serial, uint32_t time, int32_t id)
{
    if (data == NULL || touch == NULL)
        return;

    CogWlSeat *seat = data;

    if (seat->touch_target == NULL || seat->touch.surface == NULL)
        return;
     //  ^^ here is checked
     // [...] function continues [...]
```

**Comment by clopez** [comment]
changed: make seat->touch_target a weak reference

File: platform/wayland/cog-platform-wl.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (aperezdc) [reply platform/wayland/cog-platform-wl.c:851]: Great, somehow I managed to miss that we had the check already in the code when I was reading the code yesterday. Thanks for pointing it out.

**Review by aperezdc — APPROVED** [review]

**Pull request merged**